src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1183 @@
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Check if a Java file has been correctly minimized. The file should not have compilation errors
+   * or run-time errors. The file should fail in the same way as the original file.
+   *
+   * @param file the file being checked
+   * @param classpath classpath needed to compile/run the Java file
+   * @param packageName the package that the Java file is in
+   * @param expectedOutput expected output of running the JUnit test suite
+   * @param timeoutLimit number of seconds allowed for the whole test suite to run
+   * @return true if there are no compilation and no run-time errors and the output is equal to the
+   *     expected output
+   */
+  private static boolean checkCorrectlyMinimized(
+      File file,
+      String classpath,
+      String packageName,
+      Map<String, String> expectedOutput,
+      int timeoutLimit) {
+    // Zero exit status means success.
+    if (compileJavaFile(file, classpath, packageName, timeoutLimit) != 0) {
+      return false;
+    }
+
+    // Run the Java file and get the standard output.
+    String runResult = runJavaFile(file, classpath, packageName, timeoutLimit);
+
+    // Compare the standard output with the expected output.
+    return expectedOutput.equals(normalizeJUnitOutput(runResult));
+  }
+
+  /**
+   * Compile a Java file and return the compilation exit value.
+   *
+   * @param file the file to be compiled and executed
+   * @param classpath dependencies and complete classpath to compile and run the Java program
+   * @param packageName the package that the Java file is in
+   * @param timeoutLimit number of seconds allowed for the whole test suite to run
+   * @return exit value of compiling the Java file
+   */
+  private static int compileJavaFile(
+      File file, String classpath, String packageName, int timeoutLimit) {
+    // Obtain directory to carry out compilation and execution step.
+    File executionDir = getExecutionDirectory(file, packageName);
+
+    // Command to compile the input Java file.
+    String command = "javac -classpath " + systemClassPath;
+    // Add current directory to class path.
+    command += pathSeparator + ".";
+    if (classpath != null) {
+      // Add specified classpath to command.
+      command += pathSeparator + classpath;
+    }
+
+    command += " " + file.getAbsolutePath();
+
+    // Compile specified Java file.
+    return runProcess(command, executionDir, timeoutLimit).exitValue;
+  }
+
+  /**
+   * Run a Java file and return the standard output.
+   *
+   * @param file the file to be compiled and executed
+   * @param classpath dependencies and complete classpath to compile and run the Java program
+   * @param packageName the package that the Java file is in
+   * @param timeoutLimit number of seconds allowed for the whole test suite to run
+   * @return the standard output from running the Java file
+   */
+  private static String runJavaFile(
```

Comment:
I found this very hard to follow.  It mixed together operations on the classpath and non-classpath, for example.
I'm also not clear why both "." and `dirPath` are needed on the classpath.  Couldn't you set `dirPath` to "." if it would otherwise be null?  If not, please explain.

Something like this seems clearer:

```java
  private static String runJavaFile(
      File file, String userClasspath, String packageName, int timeoutLimit) {
    // Obtain directory to carry out compilation and execution step.
    File executionDir = getExecutionDirectory(file, packageName);

    // Directory path for the classpath.
    String dirPath;
    if (file.getParentFile() != null) {
      dirPath = file.getParentFile().getPath();
    } else {
      dirPath = ".";
    }

    String classpath = systemClassPath + pathSeparator + dirPath;
    if (classpath != null) {
      // Add user-specified classpath.
      classpath += pathSeparator + classpath;
    }

    // Fully-qualified classname.
    String fqClassName = FilenameUtils.removeExtension(file.getName());
    if (packageName != null) {
      fqClassName = packageName + "." + fqClassName;
    }

    // Command to run the Java file.
    String command = "java -classpath " + classpath + " org.junit.runner.JUnitCore " + fqClassName;

    // Run the specified Java file.
    return runProcess(command, executionDir, timeoutLimit).stdout;
  }
```